dot/parachain/candidate-validation/host.go

```diff
@@ -0,0 +1,103 @@
+package candidatevalidation
+
+import (
+	"fmt"
+
+	parachainruntime "github.com/ChainSafe/gossamer/dot/parachain/runtime"
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+var logger = log.NewFromGlobal(log.AddContext("pkg", "pvf"), log.SetLevel(log.Debug))
+
+type Host struct {
+	workerPool *workerPool
+}
+
+func NewValidationHost() *Host {
+	return &Host{
+		workerPool: newValidationWorkerPool(),
+	}
+}
+
+func (v *Host) Validate(msg *ValidationTask) (*ValidationResult, error) {
+	validationCodeHash := msg.ValidationCode.Hash()
+	// performBasicChecks
+	validationErr, internalErr := performBasicChecks(&msg.CandidateReceipt.Descriptor,
+		msg.PersistedValidationData.MaxPovSize,
+		msg.PoV,
+		validationCodeHash)
+
+	if internalErr != nil {
+		return nil, internalErr
+	}
+	if validationErr != nil {
+		return &ValidationResult{InvalidResult: validationErr}, nil //nolint
+	}
+	// create worker if not in pool
+	if !v.workerPool.containsWorker(validationCodeHash) {
+		worker, err := v.workerPool.newValidationWorker(*msg.ValidationCode)
+		if err != nil {
+			return nil, err
+		}
+
+		// sanity check
+		if worker.workerID != validationCodeHash {
+			return nil, fmt.Errorf("workerID does not match validationCodeHash")
+		}
+	}
```

Comment:
I think this is worker's pool responsibility (check if the worker exists, if not then create a new one as well as the sanity check)

Reply:
I've moved is logic to the worker pool.